### PR TITLE
Back End Repeating Event

### PR DESCRIPTION
**Pull request by rhrussell**
Code Coverage (any file relating to event)  - 100%
Rubocop - no offenses

### CODE REVIEW

File: app/admin/events.rb [hunk 1]
```diff
@@ -55,5 +62,100 @@
     end
     f.actions
   end
+
+  # form controller
+  controller do
```
Comment (varvind) [inline app/admin/events.rb:67]: nitpicking, but i would probably add some comments, like just explaining whats going on, at first glance its not super obvious to someone who hasnt looked at the code before

File: app/admin/events.rb [hunk 2]
```diff
@@ -29,6 +35,7 @@
     attributes_table :title, :place, :description, :starttime, :endtime, :eventpass, :created_at, :updated_at
   end
```
Comment (varvind) [inline app/admin/events.rb:36]: did you remove "actions" from this block?

**Review by varvind — CHANGES_REQUESTED** [review]

File: app/admin/events.rb [hunk 3]
```diff
@@ -55,5 +62,100 @@
...
+                                newevent['endtime(3i)'].to_i, newevent['endtime(4i)'].to_i,
+                                newevent['endtime(5i)'].to_i)
+      weeks = newevent[:repeatweeks].to_i
+      if newevent[:repeatsunday] == '1' || newevent[:repeatmonday] == '1' || newevent[:repeattuesday] == '1' ||
+         newevent[:repeatwednesday] == '1' || newevent[:repeatthursday] == '1' || newevent[:repeatfriday] == '1' ||
+         newevent[:repeatsaturday] == '1' # repeating event creation
+        # for the first day of the repeating events creation
+        event = Event.create(title: newevent[:title], place: newevent[:place], description: newevent[:description],
+                             starttime: newstarttime, endtime: newendtime, eventpass: newevent[:eventpass])
+        (1..weeks).each do |_i|
```
Comment (varvind) [inline app/admin/events.rb:87]: This should be from 0..weeks because you need to include the current week

**Review by varvind — CHANGES_REQUESTED** [review]

**Review by varvind — APPROVED** [review]
Approved once you comment the test file

**Review by varvind — CHANGES_REQUESTED** [review]
Make sure to handle case where user picks date before the current date

**Review by keithbryant3 — APPROVED** [review]
Everything looks good as far as I can tell.

**Review by keithbryant3 — APPROVED** [review]
Everything still looks good.

File: app/admin/events.rb [hunk 4]
```diff
@@ -55,5 +62,196 @@
...
+          redirect_to '/admin/events', flash: { error: 'Event was successfully created.' }
+        else
+          # gives error if weeks is negative or zero
+          redirect_to '/admin/events/new', flash: { error: 'Error: Weeks cannot be zero/negative for repeat events.' }
+        end
+      else # singular event creation
+        # creates event
+        event = Event.create(title: newevent[:title], place: newevent[:place], description: newevent[:description],
+                             starttime: newstarttime, endtime: newendtime, eventpass: newevent[:eventpass])
+        # if !event.valid? # checks to see if event is successfully created and valid
```
Comment (varvind) [inline app/admin/events.rb:242]: get rid of commented code

File: app/admin/events.rb [hunk 5]
```diff
@@ -55,5 +62,196 @@
...
+              change = (nextdate - newstarttime).to_i # gets the difference between the dates
+              # creates the event with the change
+              event = Event.create(title: newevent[:title], place: newevent[:place], description:
+                                  newevent[:description], starttime: newstarttime + change,
+                                   endtime: newendtime + change, eventpass: newevent[:eventpass])
+            end
+            newstarttime += 7
+            newendtime += 7
+
+            # if !event.valid? # checks to see if event is successfully created and valid
```
Comment (varvind) [inline app/admin/events.rb:227]: here too get rid of the commented code

**Review by varvind — CHANGES_REQUESTED** [review]
Overall code looks good, gonna do some more heroku tests but left a few comments

File: app/admin/events.rb [hunk 6]
```diff
@@ -55,5 +62,183 @@
...
+
+      if newevent[:eventpass] == '' && newevent[:title] == '' && newevent[:place] == '' &&
+         newevent[:description] == '' && newevent['starttime(1i)'] == '' &&
+         newevent['starttime(1i)'] == '' && newevent['starttime(2i)'] == '' &&
+         newevent['starttime(3i)'] == '' && newevent['starttime(4i)'] == '' &&
+         newevent['starttime(5i)'] == '' && newevent['endtime(1i)'] == '' &&
+         newevent['endtime(1i)'] == '' && newevent['endtime(2i)'] == '' &&
+         newevent['endtime(3i)'] == '' && newevent['endtime(4i)'] == '' &&
+         newevent['endtime(5i)'] == ''
+        redirect_to '/admin/events/new', flash: { error: 'Error: Please Enter Values in Required Fields.' }
```
Comment (varvind) [inline app/admin/events.rb:87]: This should say something like 'Error: Please Enter Values in All Fields

File: spec/events_spec.rb [hunk 7]
```diff
@@ -0,0 +1,688 @@
...
+    # tests whether or not app will detect an invalid single event with missing all fields
+    it 'Single Event (Missing All Fields) - Failure' do
+      visit '/admin/events' # go to the events page
+      click_link('New Event')
+      # missing all fields
+      # make sure the event is not repeating
+      click_button('Non-Repeating Event')
+      click_on('commit')
+      # page should have this error because the event is missing all fields
+      expect(page).to have_content('Error: Please Enter Values in Required Fields.')
```
Comment (varvind) [inline spec/events_spec.rb:113]: change this to the previous comment i mentioned

**Review by varvind — CHANGES_REQUESTED** [review]
Functionality is good, fix that one small change then ill approve! Good job

**Review by keithbryant3 — APPROVED** [review]
Everything looks fine to me.

**Review by varvind — APPROVED** [review]

**Pull request merged**